schema-nft-marketplace.graphql

```diff
@@ -0,0 +1,109 @@
+# Subgraph Schema: NFT Marketplace
+# Version: 0.0.1
+
+enum Network {
+  ARBITRUM_ONE
+  ARWEAVE_MAINNET
+  AURORA
+  AVALANCHE
+  BOBA
+  BSC # aka BNB Chain
+  CELO
+  COSMOS
+  CRONOS
+  MAINNET # Ethereum Mainnet
+  FANTOM
+  FUSE
+  HARMONY
+  JUNO
+  MOONBEAM
+  MOONRIVER
+  NEAR_MAINNET
+  OPTIMISM
+  OSMOSIS
+  MATIC # aka Polygon
+  XDAI # aka Gnosis Chain
+}
+
+enum NftStandard {
+    ERC721
+    ERC721A
+    ERC1155
+}
+
+enum SaleStrategy {
+    " Strategy that executes an order at a fixed price that can be taken either by a bid or an ask. "
+    StandardSaleForFixedPrice
+    " Strategy that executes an order at a fixed price that can be matched by any tokenId for the collection. "
+    AnyItemFromCollectionForFixedPrice
+    " Strategy that executes an order at a fixed price that can be matched by any tokenId in a set of tokenIds. "
+    AnyItemInASetForFixedPrice
+    " Strategy to launch a Dutch Auction for a token where the price decreases linearly until a specified timestamp and end price defined by the seller. "
+    DutchAuction
+    " Strategy to set up an order that can only be executed by a specific address. "
+    PrivateSale
+}
+
+type Marketplace @entity {
+    id: ID!
+    name: String!
+    network: Network!
+    collectionCount: Int!
+    " Cumulative sale volume (in ETH) "
+    cumulativeSaleVolumeETH: BigDecimal!
+    " Cumulative sale volume (in USD)"
+    cumulativeSaleVolumeUSD: BigDecimal!
+    " Revenue that goes to the marketplace protocol, aka protocol fee "
+    marketplaceRevenue: BigDecimal!
+    " Revenue that goes to creator, aka royalty fee"
+    creatorRevenue: BigDecimal!
+    " Sum of marketplaceRevenue and creatorRevenue "
+    totalRevenue: BigDecimal!
+    totalTransactionCount: Int!
+}
+
+type Collection @entity {
+    " Contract address "
+    id: ID!
+    name: String!
+    symbol: String!
+    nftStandard: NftStandard!
+    " Royalty fee rate in percentage. E.g. 2.5% should be 2.5 "
+    royaltyFee: BigDecimal!
+    totalSupply: BigInt!
+    " Number of tokens for sale in the marketplace "
+    listedCount: BigInt!
+    " Floor price (in ETH) "
+    floorPriceETH: BigDecimal!
+    " Floor price (in USD) "
+    floorPriceUSD: BigDecimal!
+    cumulativeSaleCount: Int!
+    " Cumulative sale volume (in ETH) "
+    cumulativeSaleVolumeETH: BigDecimal!
+    " Cumulative sale volume (in USD)"
+    cumulativeSaleVolumeUSD: BigDecimal!
+    " Revenue that goes to the marketplace protocol, aka protocol fee "
+    marketplaceRevenue: BigDecimal!
+    " Revenue that goes to creator, aka royalty fee"
+    creatorRevenue: BigDecimal!
+    " Sum of marketplaceRevenue and creatorRevenue "
+    totalRevenue: BigDecimal!
+    totalTransactionCount: Int!
```

Comment:
i'm assuming the transaction count refers to the sale count? or would this also include erc721/1155 transfers?